### DISCUSSION TITLE
[ParamConverter]

### DISCUSSION THREAD

**Discussion by Ciloe** [symfony/symfony#47949] [Q&A] [answered]
### Symfony version(s) affected

6.2.0

### Description

Hi there, I managed to update SF 6.1 to SF 6.2. and I have an error about my ParamConverter :
```
Binding entities to query parameters only allowed for entities that have an identifier.
Class "App\Entity\Event" does not have an identifier.
```

### How to reproduce

I did this in my ParamConverter :
```php
                $request->attributes->set(
                    $configuration->getName(),
                    $parent->getOccurrence($date)
                );
                return true;
```
and the entity Event has no id because it does not exist in database. But I don't want doctrine going to work on my paramConverter. What's the matter ?

**Comment by xabbuh** [comment]
Can you create a small example application that allows to reproduce your issue?

**Comment by derrabus** [comment]
Moving to support

**Comment by MatTheCat** [comment]
> the entity Event has no id because it does not exist in database

So that's not an entity. Do not tell Doctrine it's one and the DoctrineParamConverter won't be called.

**Reply by Ciloe** [reply]
That's not the question, when I use a custom param converter, I don't wan't that the doctrine param converter is call. It's working fine in the 6.1 version and it's broken in the 6.2 version....

**Reply by MatTheCat** [reply]
> I don't wan't that the doctrine param converter is call.

> Do not tell Doctrine [`Event` is an entity] and the DoctrineParamConverter won't be called.

How isn’t it the question? :thinking: 

Anyway if you don’t want to fix your mapping you can specify your ParamConverter’s name in the annotation or increase its priority (by default every converter has the same so their order is not guaranteed. That’s what you observed.)

BTW all of this is mentioned in [the SensioFrameworkExtraBundle documentation](https://symfony.com/bundles/SensioFrameworkExtraBundle/current/annotations/converters.html#usage).

**Reply by Ciloe** [reply]
> you can specify your ParamConverter’s name

Yes, I deed that

```php
#[ParamConverter('event', class: Event::class, converter: 'occurrence_converter')]
public function eventEdit(
        Request $request,
        Establishment $establishment,
        EditEvent $eventService,
        Event $event
    )
```

> increase its priority

I did to : 

```yaml
  App\Service\Common\OccurrenceParamConverter:
    tags:
      - { name: 'request.param_converter', converter: 'occurrence_converter', priority: 100 }
```

But it's not working

**Comment by Ciloe** [comment]
I will create an example project when I have the time. Thanks !

**Reply by derrabus** [reply]
Yes, that would be helpful. Or at least give us a stack trace of the error.

**Comment by Ciloe** [comment, marked as answer]
I just migrate to SF 6.2 and I haven't this error. It's working fine now :)
